modules/45_carbonprice/NDC/datainput.gms

```diff
@@ -15,64 +16,99 @@ pm_taxCO2eq("2020",regi)= pm_taxCO2eq("2020",regi) * sm_DptCO2_2_TDpGtC;
+Scalar p45_use_single_year_close_to      "if 0: use all. If > 0: use only one single NDC target per country closest to this year (use 2030.4 to prefer 2030 over 2035 over 2025)" /2030.4/;
+
+Set p45_NDC_year_set(ttot,all_regi)               "YES for years whose NDC targets is used";
+Parameter p45_best_NDC_coverage(all_regi)         "highest coverage of NDC targets within region";
+Parameter p45_distance_to_optyear(ttot,all_regi)  "distance to p45_use_single_year_close_to to favor years in case of multiple equally good targets";
+Parameter p45_min_distance_to_optyear(all_regi)   "minimal distance to p45_use_single_year_close_to per region";
+
+p45_best_NDC_coverage(regi) = smax(ttot$(ttot.val <= p45_ignore_NDC_after AND ttot.val >= p45_ignore_NDC_before), p45_2005share_target(ttot,regi));
+display p45_best_NDC_coverage;
+
+p45_NDC_year_set(ttot,regi)$(ttot.val <= p45_ignore_NDC_after AND ttot.val >= p45_ignore_NDC_before) = p45_2005share_target(ttot,regi) >= p45_min_ratio_of_coverage_to_max * p45_best_NDC_coverage(regi);
+
+if(p45_use_single_year_close_to > 0,
+  p45_distance_to_optyear(p45_NDC_year_set(ttot,regi)) = abs(ttot.val - p45_use_single_year_close_to);
+  p45_min_distance_to_optyear(regi) = smin(ttot$(p45_NDC_year_set(ttot,regi)), p45_distance_to_optyear(ttot,regi));
+  p45_NDC_year_set(ttot,regi) = p45_distance_to_optyear(ttot,regi) = p45_min_distance_to_optyear(regi);
+);
+
+*** first and last NDC year as a number
+Parameter p45_first_NDC_year(all_regi) "last year with NDC coverage within region";
+p45_first_NDC_year(regi) = smin( p45_NDC_year_set(ttot, regi), ttot.val );
+Parameter p45_last_NDC_year(all_regi)  "last year with NDC coverage within region";
+p45_last_NDC_year(regi)  = smax( p45_NDC_year_set(ttot, regi), ttot.val );
+
+display p45_NDC_year_set,p45_first_NDC_year,p45_last_NDC_year;
+
+*** adjust reduction value for LAM based on the assumption that Brazilian reduction targets are only from landuse, see https://climateactiontracker.org/countries/brazil/
+*** the adjustment were calculated such that Brazil is assumed to maintain its 2015 non-landuse emissions, as follows:
+*** Use R and the code in https://github.com/pik-piam/mrremind/blob/master/R/calcEmiTarget.R to calculate dummy1, ghgTarget, ghgfactor, then run the following code:
+*** countries <- toolGetMapping("regionmappingREMIND.csv",where = "mappingfolder",type = "regional")
+*** LAMCountries <- countries$CountryCode[countries$RegionCode == "LAM"]
+*** shareWithinTargetCountries <- dummy1[LAMCountries,"y2030",] * ghgTarget[LAMCountries,"y2030",] / dimSums(dummy1[LAMCountries,"y2030",] * ghgTarget[LAMCountries,"y2030", ], dim=1)
+*** print(shareWithinTargetCountries["BRA",,]*(as.numeric(ghg["BRA","y2015"])/as.numeric(ghg["BRA","y2005"])-as.numeric(ghgfactor["BRA","y2030","gdp_SSP2"])))
+*** 0.2 is a rounded value valid for all except 2018_uncond, because Brazil had no unconditional target then.
```

Comment:
could you also delete those lines?

Reply:
well, this is the detailed explanation you desired on how I determined the 0.2…

Reply:
sorry, mixed it up